scripts/metrics/common/config.py

```diff
@@ -0,0 +1,79 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Config file utilities for metrics scripts
+"""
+
+import os
+import sys
+
+import yaml
+
+from common.exception import MetricsExn
+
+config = {}
+
+
+class ConfigExn(MetricsExn):
+    pass
+
+
+def load_config(local_path, override=None):
+    global config
+
+    if override is None:
+        override = {}
+
+    try:
+        with open(os.path.join(sys.path[0], local_path), "r") as file:
+            config = yaml.safe_load(file)
+    except FileNotFoundError as exc:
+        raise ConfigExn(f"Failed to load configuration from local path '{local_path}'") from exc
+
+    for path, value in override:
+        path_elements = path.split(".")
+        target = config
+
+        while len(path_elements) > 0:
+            key = path_elements.pop(0)
+
+            if isinstance(target, dict) and key in target:
```

Comment:
This is primarily used to for both local testing + being automation accessible. It's simple for a local user to set their own configuration while testing. The override is used for pipelines to paramterize different configuration (such as jenkins base urls).

Since this is meant to override existing keys, this should only be updating and not adding.

I'm unsure what you mean by this. I think there is some confusion that doc strings should clarify. It's kind of like multi layered keys with a value i.e.
Key1.Key2 = value
Key1.Key3 = value

Reply:
I don't see why it needs to thread through this data-transformer script.
1. If I'm going to add data to a file, then I'll the data to the source of truth. Users can directly modify an XML, or directly modify a CSV after this performs transformation.
2. If I'm going to expect/query different data, then I'll access different data. Tests can directly feed in a different CSV data.

Reply:
Removed config util file

Reply:
What I'm getting at here is essentially the https://en.wikipedia.org/wiki/Single-responsibility_principle and https://en.wikipedia.org/wiki/Interface_segregation_principle of SOLID